### ISSUE TITLE
Give the user hints on how to solve failures.

### ISSUE THREAD

**Issue by SvenDowideit** [docker/machine#3765]
If there is a failure when creating a VM, chances are, that trying to `rm` it will also fail, because the creation left things in an inconsistent state.

in that case, it'd be good to tell them things like `add "-f" to "machine rm" to force it to delete`

looks like one `create`  failed due to out of disk space - but i can only find that out by doing an `rm -f`, and then re-running the create with `--debug` :(

This kind of helpful UX would make this not-so-often used tool much easier to work with

or similarly, when the creation fails, mark the vm config that you store as "creation may have failed"

TBH, it'd be even more sensible to not mark a vm as "working" until the creation has succeeded, and connection to the vm has been tested as good.

**Comment by nathanleclaire** [comment]
Yes, it's motivation behind tickets like https://github.com/docker/machine/issues/2712 and https://github.com/docker/machine/issues/1821.
